buildpack/v05/encoderdecoder.go

```diff
@@ -0,0 +1,42 @@
+package v05
+
+import (
+	"os"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/buildpack/layertypes"
+)
+
+type EncoderDecoder05 struct {
+}
+
+func NewEncoderDecoder() *EncoderDecoder05 {
+	return &EncoderDecoder05{}
+}
+
+func (d *EncoderDecoder05) IsSupported(buildpackAPI string) bool {
+	return api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) < 0
+}
+
+func (d *EncoderDecoder05) Encode(file *os.File, lmf layertypes.LayerMetadataFile) error {
+	return toml.NewEncoder(file).Encode(lmf)
+}
+
+func (d *EncoderDecoder05) Decode(path string) (layertypes.LayerMetadataFile, string, error) {
+	var lmf layertypes.LayerMetadataFile
```

Comment:
I think I would rather see a private struct here to decode into and map to `layertypes.LayerMetadataFile` like we do in the `06` encoder. It's not a big deal, but this decode path going directly to the resulting struct means we could introduce new fields and the `05` users would be able to populate those fields.

Reply:
I totally agree with what you're suggesting but I'm wondering whether we should introduce this struct now or whether we should wait for the time when we need to add new fields.
I lean a bit to the later since if it takes time until we introduce new fields, someone who isn't familiar with the code might wonder why we add this redundant struct.
What do you think?

Reply:
I don't have a strong opinion on this, but I personally lean toward creating explicit structs for each version and treating the `layertypes.LayerMetadataFile` as the internal interface we code against outside of parsing/decoding/encoding. My main concern is around not remembering that `EncoderDecoder05` is _directly_ decoding to the internal interface.  But again, no strong preference.

Reply:
Following https://github.com/buildpacks/lifecycle/pull/537/commits/2373122d4321c49d25e9d495e98b6755013b4095 and a discussion I had with @jromero, I'm wondering whether we want to keep `buildpack/files.go` as is or split it so each go file will be related to a different file.
If we do it, in order to prevent circular imports, we'll need to redefine the `EncoderDecoder` interface in `v05/encoderdecorder.go` and `v06/encoderdecorder.go`, and we'll need to define `LayerMetadataFile` in `v05/encoderdecorder.go`.
@jabrown85 @natalieparellano, what do you think?
@natalieparellano, during the refactoring you did recently, I remember that there was a reason that you didn't split `buildpack/files.go` but I don't remember what it was. I guess it can be related to our decision.

Reply:
I think @natalieparellano recently had a change in opinion on not splitting them up in the first place. As far as the `EncoderDecoder` interface, it should be in a different package than the version specific implementations. The interfaces can live under `lifecycle` in a single place perhaps. Sort of like docker does [here](https://github.com/moby/moby/blob/master/client/interface.go) with the `client` package. All the interfaces living in `lifecycle` will allow the `test` and `vXX` packages to reference them. It also makes it very obvious if a `vXX` package is relying on another `vXX` package instead of the defined interface.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/d6aa6e72ae980771d30efd95383086f0e1995f67.